### DISCUSSION TITLE
Using yauaa to distinguish browser and thick app

### DISCUSSION THREAD

**Discussion by yyy** [nielsbasjes/yauaa#382] [Q&A] [answered]
I have a usecase where I want to distinguish requests that come from browsers vs thick apps. I need this for both desktop and mobile, but I was specifically testing on iOS and Android. I've observed that on Android, I'm receiving AgentType "Browser" for browsers, and "Browser WebView" for thick applications that embed WebView. But on iOS, AgentType comes as "Browser" for both.

What is the recommended way to find out if the request is coming from a browser or a thick app?

**Comment by nielsbasjes** [comment]
Do you have examples for all of these?

**Reply by yyy** [reply]
Sorry it took so long to respond. It looks like the latest version of yauaa is better than when I tried it...

Here are some user agent strings that we've collected:
```
SalesforceMobileSDK/9.1.1 iOS/14.8.1 (iPhone) Chatter/234.020(6203393) Hybrid uid_9325236B-A894-45E2-BA6D-F9B37FFC8423 ftr_ Mozilla/5.0 (iPhone; CPU iPhone OS 14_8_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 iPhone/14.8.1 iOS_iPhone Salesforce1/234.020(6203393)
Mozilla/5.0 (iPhone; CPU iPhone OS 14_8_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.2 Mobile/15E148 Safari/604.1
Mozilla/5.0 (iPhone; CPU iPhone OS 14_8_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148

SalesforceMobileSDK/9.1.0 android mobile/11 (Pixel 3 XL) Salesforce1/234.020.0(234020000) Native uid_fcfde120405e320d ftr_ SecurityPatch/2021-10-01 Cordova/8.0.0 auraff/sm Mozilla/5.0 (Linux; Android 11; Pixel 3 XL Build/RQ3A.211001.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/96.0.4664.92 Mobile Safari/537.36
Mozilla/5.0 (Linux; Android 11; Pixel 3 XL Build/RQ3A.211001.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/96.0.4664.92 Mobile Safari/537.36 PKeyAuth/1.0
Mozilla/5.0 (Linux; Android 11; Pixel 3 XL) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.92 Mobile Safari/537.36
```

These are all using a browser component inside mobile apps, but I understand that for some mobile apps the user-agent is identical to the browser user agent, and YAUAA won't be able to distinguish them simply based on the User-Agent.

The only one that goes seriously wrong is SalesforceMobile on iOS where AgentClass is reported as Browser (should be something else).

**Reply by nielsbasjes** [reply]
In the SalesforceMobileSDK on iOS (the Chatter thing) you see that it is classified as a Webview.
This is essentially a webbrowser which was embedded in an app.
So yes I agree: "Browser" doesn't seem right.

**Reply by nielsbasjes** [reply]
I improved this. Please check https://try.yauaa.basjes.nl/ if this is better.

**Comment by nielsbasjes** [comment, marked as answer]
I'm assuming this has been answered.
